### PR TITLE
update cron schedule parse and unit test

### PR DESCRIPTION
**Pull request by Metrora**
Signed-off-by: Liucw <liu.chuwei@99cloud.net>

### What type of PR is this?
/kind feature

### What this PR does / why we need it:
update the way to parse month-end cron schedule

### Which issue(s) this PR fixes:

Fixes https://github.com/kubeclipper-labs/kubeclipper/issues/177

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```
@x893675

### CODE REVIEW

File: pkg/controller/cronbackupcontroller/controller_test.go [hunk 1]
```diff
@@ -22,13 +26,54 @@ func Test_parseSchedule(t *testing.T) {
...
+				signer: &signer{
+					now: func() time.Time {
+						return time.Date(2022, 1, 1, 0, 0, 0, 0, time.UTC)
+					},
+				},
 			},
 			want: "0 0 31 * *",
 		},
+		{
+			name: "test parse cron schedule with day of month is 'L'",
```
Comment (lixd) [inline pkg/controller/cronbackupcontroller/controller_test.go:38]: change test name

File: pkg/controller/cronbackupcontroller/controller.go [hunk 2]
```diff
@@ -69,6 +69,27 @@ type CronBackupReconciler struct {
...
+	now func() time.Time
+}
+
+func newSigner() *signer {
+	return &signer{
+		now: time.Now,
+	}
+}
+
+func parseSchedule(schedule string, signer *signer) string {
```
Comment (lixd) [inline pkg/controller/cronbackupcontroller/controller.go:82]: func(signer *signer) parseSchedule(schedule string)

**Review by lixd — APPROVED** [review]
/lgtm

**Comment by x893675** [comment]
/retitle update cron schedule and unit test

**Comment by x893675** [comment]
/retitle update cron schedule parse and unit test

**Pull request merged**